### DISCUSSION TITLE
[Help Wanted] FileSafe Migration Google Drive Auth

### DISCUSSION THREAD

**Discussion by Ed-V** [standardnotes/forum#2936] [General]
**Describe the issue**
Attempting to add Google Drive as integration directs me to an error page and I am unable to authenticate. I need to authenticate to migrate old files.

![F1BE277E-AB19-4B1D-AC74-49D8B71EE1DB](https://user-images.githubusercontent.com/10635618/212392093-ae1c5ca5-e9fe-4dc9-921d-700e25ab68cd.jpeg)

**To Reproduce**
1. Download the Filesafe addon https://extensions.standardnotes.org/repos/legacy/filesafe
2. Go through the steps to enable it in quick menu
3. Go to integration and click on add new and try to link google drive 

**Expected behavior**
To authenticate successfully and be able to migrate my old files

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser Firefox 
 - Version 107.0.1

**Comment by xanoni** [comment]
Maybe related: https://github.com/standardnotes/forum/issues/1202

**Comment by xanoni** [comment]
I have the same / a similar issue with Dropbox now:

![image](https://user-images.githubusercontent.com/77220130/212251466-d60cd165-2112-413f-94d9-65b7d606dfe6.png)

**Comment by jmason888** [comment]
Same issue here.  I thought there was no hurry to transition my files out of FileSafe...

**Comment by xanoni** [comment]
> Same issue here. I thought there was no hurry to transition my files out of FileSafe...

Same... but seems that was wrong information on the SN blog. Of course the integrations would break eventually, and seems they're no longer planning to upgrade them. SN team gotta clean this up, please.

**Comment by effieeee** [comment]
Hi all, apologies for the wait, but the 400 errors should be fixed now for both FileSafe and Cloud Backups🙏

**Comment by Massis1** [comment]
Hello,
filesafe no longer has the keys and does not work; Can't download files from google
How can I download and decrypt 2 files that I forgot to convert?

**Comment by xanoni** [comment]
@effieeee so just to confirm, is the below the current state of things?
* Dropbox can still be linked/authenticated as a backup source, but it's no longer possible for the app to download encrypted FileSafe files from Dropbox
* However, I am still able to decrypt the files by downloading them from Dropbox and dragging them into the FileSafe window

If yes, then at least I can access my files again, even though I'll have to manually sort through hundreds of files and figure out where they belong / what they are / why I put them in SN. (Most of it is probably random screenshots that I took at some point and pasted into one of the editors.)

**Questions:**
1. Can you please confirm / clarify the above?
2. Are you planning to make the Dropbox integration functional again (e.g., such that inline images would display in the editors again), or should I decrypt my files now and migrate them somewhere else?
3. If the latter, is there still support for inline images using some other backend?
4. Will Dropbox still be supported as a backup location going forward (I mean for notes backups, not for attachments)?
5. Are the answers to above questions the same for the other two external integrations (Google Drive and OneDrive)?

Thank you!

**Comment by xanoni** [comment]
Okay, so I'm indeed able to decrypt my attachment files from Dropbox by downloading them and then dragging them into the FileSafe window. Relief, the worst case (data loss) has been prevented.

**But is there some way to perform the decryption efficiently for almost 1,000 encrypted files?**

I tried dragging the whole folder into the FileSafe window, but the app would just show the "Reading file" message and get stuck. I waited for a day, but there was never any noticeable disk/CPU activity.

Thus, I assume that dragging folders is simply not supported at this point, and that the app doesn't handle this exception properly (no error message).

![image](https://user-images.githubusercontent.com/77220130/216810239-b35fafb1-afd9-46df-93eb-026d302ab92f.png)

When I went into the folder and dragged multiple encrypted files into FileSafe at once, SN only decrypted a single one of them. It would likely take me hours to decrypt those ~1,000 files one-by-one and my brain would die off in the process. Most files have the same filename, `image.png`, so I would need to drag files, go into the filename field, adjust the names, etc... almost 1,000 times. It's also very error prone.

**Questions:**
_[... my questions 1-5 from above are still relevant ...]_
6. Are you planning to make the GUI more efficient to use?
7. If not, is there a batch/CLI way to achieve the same task?

Thank you!

**Comment by xanoni** [comment]
By the way, even after I have decrypted all those files, the situation is not going to be very nice ... those were mostly inline images with generic filenames (`image.png`), so I would have to manually go through them 1-by-1 and re-attach them to their respective notes again.

I just checked out a few examples of notes with inline images, and they now say something like the below instead.

```
Unable to download image.png XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX.
```

However, when I search for this ID string in my folder of encrypted attachments via the below, it doesn't find anything.

```
find . -iname 'XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX*'
[... nothing ...]

grep -ir 'XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX' .
[... nothing ...]
```

I was initially concerned that FileSafe might have never uploaded all of my attachments to Dropbox, but next I found that the ID shown in the note is _different_ from the one shown for the same file below the note, in the FileSafe section. (The ID is shown when I click one of the filenames.)

And worse, I can't even copy/paste that second ID from the GUI.

This is all very messed up. There must be a better migration process than this.
